Mathlib/Data/Nat/Defs.lean

```diff
@@ -670,6 +670,15 @@ protected lemma div_pow (h : a ∣ b) : (b / a) ^ c = b ^ c / a ^ c := by
   refine (Nat.div_eq_of_eq_mul_right (pos_pow_of_pos c ha) ?_).symm
   rw [← Nat.mul_pow, Nat.mul_div_cancel_left' h]
 
+lemma pow_self_pos : ∀ n : ℕ, 0 < n ^ n
```

Comment:
Can we also have a lemma describing exactly when m ^ n is positive (ie Nat.pow_pos_iff : 0 < m ^ n iff m > 0 or n = 0)? Then this one will be `simp [Nat.pow_pos_iff]`